python/dgl/graphbolt/impl/csc_sampling_graph.py

```diff
@@ -855,5 +858,5 @@ def from_dglgraph(
             type_per_edge,
             edge_attributes,
         ),
-        metadata,
+        None if is_homogeneous else metadata,
```

Comment:
add `metadata=None` above line 831. then we don't need if-else check here again.